### PR TITLE
Update BioAlignments and BioSequences compat. Change LongAminoAcidSeq to LongSequence{AminoAcidAlphabet}

### PR DESCRIPTION
**Pull request by kool7d**
Update BioAlignments and BioSequences compat in Project.toml. 
Change LongAminoAcidSeq to LongSequence{AminoAcidAlphabet}. (new type)

## Types of changes

This PR implements the following changes:
_(Please tick any or all of the following that are applicable)_

* [ ] :sparkles: New feature (A non-breaking change which adds functionality).
* [x] :bug: Bug fix (A non-breaking change, which fixes an issue).
* [ ] :boom: Breaking change (fix or feature that would cause existing functionality to change).

## :clipboard: Additional detail

  - **Does your change alter APIs or existing exposed methods/types?**

Change LongAminoAcidSeq to LongSequence{AminoAcidAlphabet}. (new type)

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -21,10 +21,10 @@ RecipesBase = "3cdcf5f2-1ef4-517c-9805-6587b60abb01"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 
 [compat]
-BioAlignments = "2"
+BioAlignments = "2, 3"
```
Comment (jgreener64) [inline Project.toml:24]: Drop the `2` in each case.

File: src/model.jl [hunk 2]
```diff
@@ -89,7 +89,7 @@ export
     disorderselector,
     hydrogenselector,
     allselector,
-    LongAminoAcidSeq,
+    LongSequence{AminoAcidAlphabet},
```
Comment (jgreener64) [inline src/model.jl:92]: No type parameter on the export?

**Review by jgreener64 — CHANGES_REQUESTED** [review]
Thanks for doing this. I must admit I was putting it off because I think it requires a v2 tag for BioStructures.jl, but I guess it is time to bite the bullet.

There are a few other cases of `LongAminoAcidSeq` that need to be changed too: in src/mmtf.jl, src/spatial.jl and test/runtests.jl.

File: Project.toml (hunk 1 above)
Comment (kool7d) [reply Project.toml]: BioCore is a unique one. I think it is still v2 actually. Hasn't been updated in a very long time (5 years)

**Comment by kool7d** [comment]
Looks like a new test error unrelated to the specific changes? 
pdbentrylist() finds an invalid pdb id "     HYD".
Doesn't make sense since it is longer than 4 characters, and I searched the "ftp://ftp.wwpdb.org/pub/pdb/derived_data/index/entries.idx" doc and can't find it.

**Comment by jgreener64** [comment]
Great, thanks for this.

**Pull request merged**